### PR TITLE
`wallet.dumpPrivKey` and `wallet.importPrivKey` RPC 

### PR DESCRIPTION
**Pull request by siradji**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
RPC wallet implementation #48 
#### Which issue(s) does this PR fixes?:

Fixes #

### CODE REVIEW

File: website/docs/jellyfish/api/wallet.md [hunk 1]
```diff
@@ -279,3 +279,23 @@ enum Mode {
   CONSERVATIVE = 'CONSERVATIVE'
 }
 ```
+
+## dumpPrivKey
+
+Reveals the private key corresponding to 'address'. Then the importprivkey can be used with this output.
```
Comment (fuxingloh) [inline website/docs/jellyfish/api/wallet.md:285]:
```suggestion
Reveals the private key corresponding to an address.
```

File: packages/jellyfish-api-core/src/category/wallet.ts [hunk 2]
```diff
@@ -249,6 +249,28 @@ export class Wallet {
       'bignumber'
     )
   }
+
+  /**
+   * Reveals the private key corresponding to 'address'.
+   *
+   * Then the importprivkey can be used with this output.
```
Comment (fuxingloh) [inline packages/jellyfish-api-core/src/category/wallet.ts:256]:
```suggestion
  /**
   * Reveals the private key corresponding to an address.
   *
```

**Review by fuxingloh — CHANGES_REQUESTED** [review]

**Review by canonbrother — COMMENTED** [review]
overall looks good.. but we have a new test structure.. gonna refactor a bit

**Comment by siradji** [comment]
> overall looks good.. but we have a new test structure.. gonna refactor a bit

@canonbrother, i have implemented that.

File: packages/jellyfish-api-core/__tests__/category/wallet.test.ts [hunk 3]
```diff
@@ -876,3 +876,157 @@ describe('masternode', () => {
...
+  describe('regTest', () => {
+    const container = new RegTestContainer()
+    const client = new ContainerAdapterClient(container)
+
+    beforeAll(async () => {
+      await container.start()
+      await container.waitForReady()
+    })
+
+    afterAll(async () => {
```
[the remark this answers is not in the document]
Comment (siradji) [reply packages/jellyfish-api-core/__tests__/category/wallet.test.ts]: @fuxingloh , That's is done.

**Review by fuxingloh — DISMISSED** [review]

File: packages/jellyfish-api-core/__tests__/category/wallet/importPrivkey.test.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should should import private key without failing ', async () => {
```
Comment (jingyi2811) [inline packages/jellyfish-api-core/__tests__/category/wallet/importPrivkey.test.ts:18]: Extra should

File: website/docs/jellyfish/api/wallet.md [hunk 5]
```diff
@@ -279,3 +279,23 @@ enum Mode {
...
 }
 ```
+
+## dumpPrivKey
+
+Reveals the private key corresponding to an address.
+
+```ts title="client.wallet.dumpPrivKey()"
+interface wallet {
+  dumpPrivKey(address: string): Promise<string>
```
Comment (jingyi2811) [inline website/docs/jellyfish/api/wallet.md:289]: Add a space between dumpPrivKey and (address: string): Promise<string>

File: website/docs/jellyfish/api/wallet.md [hunk 6]
```diff
@@ -279,3 +279,23 @@ enum Mode {
...
+}
+```
+
+## importPrivKey
+
+Adds a private key (as returned by dumpprivkey) to your wallet. Requires a new wallet backup.
+
+```ts title="client.wallet.importPrivKey()"
+interface wallet {
+  importPrivKey(privkey: string, label: string = "", rescan: boolean = true): Promise<void>
```
Comment (jingyi2811) [inline website/docs/jellyfish/api/wallet.md:299]: Same

**Review by jingyi2811 — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**